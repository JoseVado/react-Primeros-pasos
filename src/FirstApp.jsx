import PropTypes from 'prop-types';


/*const newMesage = {
  mesage: "Hola Mundo",
  title: "Vado",
};

const mensaje = () => "Es el retorno de una funciÃ³n";
*/


export const FirstApp = ({ title, subTitle }) => {
    
    /*if (!title) {
        throw new Error('El title no existe');
    }*/

  return (
    <>
      <h1> {title} </h1>
      {/*<h1>{ JSON.stringify( newMesage )}</h1>*/}
      <p> {subTitle} </p>
    </>
  );
};

FirstApp.propTypes = {
  title: PropTypes.string.isRequired,
  subTitle: PropTypes.string.isRequired,
};

FirstApp.defaultProps = {
    title: 'No hay titulo',
    subTitle: 'No hay subtitulo',
}
