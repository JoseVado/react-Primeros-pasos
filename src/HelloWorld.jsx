

export const HelloWorld = () => {
    return (
        <h1>HelloWorld App</h1>
    );
}
