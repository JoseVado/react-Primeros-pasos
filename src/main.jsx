import React from 'react'
import ReactDOM from 'react-dom/client';

{
  /*import { HelloWorld } from './HelloWorld'
import { FirstApp } from './FirstApp';*/
}

import {CounterApp} from './CounterApp';

import './styles.css';

ReactDOM.createRoot(document.getElementById("root")).render(
  <React.StrictMode>
    {/*<HelloWorld />*/}
    {/*title="hola goku" subTitle={123}*/}
        {/*<FirstApp title="Hola soy goku" subTitle={123} />*/}
    <CounterApp value={2} />
  </React.StrictMode>
);