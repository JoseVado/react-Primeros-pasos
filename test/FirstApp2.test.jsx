import { render,screen } from "@testing-library/react";
import { FirstApp } from "../src/FirstApp";


describe('pruebas en <FirstApp />', () => {

    const titulo = "Hola soy goku";
    const subtitulo = "Soy un subtitulo";
    
    test('debe de hacer match con el snahpshot', () => {
        
        const {container} = render(<FirstApp title={titulo} />);
        expect(container).toMatchSnapshot();
    });

    test('Debe de mostrar el mensaje "Hola soy Goku"', () => {

        render(<FirstApp title={titulo} />);
        expect( screen.getByText(titulo) ).toBeTruthy();

    });

    test('Debe de mostrar el titulo en un h1', () => {

        render(<FirstApp title={titulo} />);
        expect(screen.getByRole('heading', { level: 1 }).innerHTML).toContain(titulo);
      
    });
    
    test('Debe de mostrar el subtitulo enviado por props', () => {

        render(
          <FirstApp title={titulo} subTitle={subtitulo} />
        );

        expect(screen.getAllByText(subtitulo).length).toBe(2);
    });
});