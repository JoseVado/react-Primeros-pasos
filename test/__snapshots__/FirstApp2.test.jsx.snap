// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <FirstApp /> debe de hacer match con el snahpshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
     
    Hola soy goku
     
  </h1>
  <p>
     
    No hay subtitulo
     
  </p>
  <p>
     
    No hay subtitulo
     
  </p>
  <p>
     
    Vado Robles
     
  </p>
</div>
`;
